appserver/extras/payara-micro/payara-micro-boot/src/main/java/fish/payara/micro/boot/PayaraMicroBoot.java

```diff
@@ -86,6 +87,25 @@
      * @return
      */
     PayaraMicroBoot addRepoUrl(String... URLs);
+    
+    /**
+     * Adds a command to run before booting Payaa Micro.
+     * Not that only a limited set of commands will run here i.e. {@code set}.
+     * @param command Command to add
+     */
+    public void addPreBootCommand(BootCommand command);
+    
+    /**
+     * Adds a command to run after Payara Micro has booted
+     * @param command Command to add
+     */
+    public void addPostBootCommand(BootCommand command);
+    
+    /**
+     * Adds a command to be run after the application is deployed
+     * @param command Command to add
+     */
+    public void addPostDeployCommand(BootCommand command);
```

Comment:
PostDeploy is equivalent `run`, so it likely doesn't need explicit api.

Only if we'd deprecate `run` forms that execute asadmin, because they are mixed with data processing.